internal/integration/bounce_rule_test.go

```diff
@@ -0,0 +1,211 @@
+package integration
+
+import (
+	"bytes"
+	"encoding/json"
+	"net/http"
+	"testing"
+
+	"github.com/go-sql-driver/mysql"
+
+	"github.com/jimmyjames85/bouncecm/internal/models"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+)
+
+type BounceRuleSuite struct {
+	suite.Suite
+
+	handler http.HandlerFunc
+	client  *http.Client
+}
+
+func (suite *BounceRuleSuite) SetupSuite() {
+	suite.client = &http.Client{}
+}
+
+func (suite *BounceRuleSuite) SetupTest() {
+	_, err := Database.Exec(`
+		CREATE TABLE IF NOT EXISTS bounce_rule (
+			id smallint(5) unsigned NOT NULL AUTO_INCREMENT,
+			response_code smallint(5) unsigned NOT NULL DEFAULT '0',
+			enhanced_code varchar(16) NOT NULL DEFAULT '',
+			regex varchar(255) NOT NULL DEFAULT '',
+			priority tinyint(3) unsigned NOT NULL DEFAULT '0',
+			description varchar(255) DEFAULT NULL,
+			bounce_action varchar(255) NOT NULL,
+			PRIMARY KEY (id),
+			UNIQUE KEY bounce_rule_components (response_code,enhanced_code,regex)
+		) ENGINE=InnoDB DEFAULT CHARSET=latin1;
+	`)
+
+	if err != nil {
+		suite.T().Fatalf("Failed to setup for test\nError: %v", err)
+	}
+
+	mysql.RegisterLocalFile("testdata/bounce_rules.csv")
+
+	res, err := Database.Exec("LOAD DATA LOCAL INFILE '" + "testdata/bounce_rules.csv" + "' INTO TABLE bounce_rule FIELDS TERMINATED BY ',' LINES TERMINATED BY '\n'")
+	if err != nil {
+		suite.T().Fatalf("Failed to load data from file\nError: %v", err)
+	}
+	inserted, err := res.RowsAffected()
+	if err != nil {
+		suite.T().Fatalf("Failed to get inserted rows\nError: %v", err)
+	}
+	if inserted != 295 {
+		suite.T().Fatalf("Expected %d rows in bounce_rule table, got %d", 295, inserted)
+	}
+}
+
+func (suite *BounceRuleSuite) TestGetAllBounceRulesHandler() {
+	req, err := http.NewRequest("GET", "http://localhost:4000/bounce_rules", nil)
+	if err != nil {
+		suite.T().Fatalf("Error in forming request\nError: %v", err)
+	}
+
+	res, err := suite.client.Do(req)
+	if err != nil {
+		suite.T().Errorf("Failed to execute request\nError: %v", err)
+	}
+	if status := res.StatusCode; status != http.StatusOK {
+		suite.T().Errorf("Handler returned wrong status code: got %v want %v", status, http.StatusOK)
+	}
+
+	assert.Equal(suite.T(), http.StatusOK, res.StatusCode)
+}
+
+func (suite *BounceRuleSuite) TestGetSingleBounceRuleHandler() {
```

Comment:
Nice job testing both found and not found cases. Again, try to add in validation of the response too.